srml/executive/src/lib.rs

```diff
@@ -114,6 +114,9 @@ mod internal {
 				DispatchError::Stale => ApplyError::Stale,
 				DispatchError::Future => ApplyError::Future,
 				DispatchError::BadProof => ApplyError::BadSignature(""),
+				DispatchError::BlockExhausted => ApplyError::Future,
```

Comment:
The same question: what is the difference between `::Resource` and `::BlockExhausted`.

I couldn't find any logic based on the error type except for the `basic_authorship.rs` retrying a few times, when it receives a `FullBlock` back, before quitting. `::Future` doesn't seem to re-schedule.